common/test/acceptance/tests/lms/test_lms_problems.py

```diff
@@ -954,3 +954,71 @@ def test_grader_type_displayed(self):
         self.assertEqual(problem_page.problem_name, 'TEST PROBLEM')
         self.assertEqual(problem_page.problem_progress_graded_value, "1 point possible (ungraded)")
+
+
+class ProblemNeverShowCorrectnessTest(ProblemsTest):
+    """
+    Ensure that correctness is not leaked when showing problems where show_correctness=never.
+    """
+    def get_problem(self):
+        """
+        Creates a {problem_type} problem with show_correctness="never"
+        """
+        # Generate the problem XML using capa.tests.response_xml_factory
+        return XBlockFixtureDesc(
+            'problem',
+            self.problem_name,
+            data=self.factory.build_xml(**self.factory_kwargs),
+            metadata={'rerandomize': 'always', 'show_correctness': 'never'},
+        )
+
+    def assert_correctness_withheld(self):
+        """
+        Ensure correctness is withheld when answer is submitted.
+        """
+        self.problem_page.click_submit()
+        self.wait_for_status('answered')
+        self.problem_page.wait_answered_notification()
+        self.assertEqual('Question 1: answer received', problem_page.status_sr_text)
+
+    @attr(shard=7)
+    def test_answer_correct_withheld(self):
+        """
+        Scenario: I can answer a problem correctly, and correctness is withheld.
+        Given External graders respond "correct"
+        and I am viewing a "<ProblemType>" problem that shows the correctness "never"
+        When I answer a "<ProblemType>" problem "correctly"
+        Then my "<ProblemType>" answer is marked "answered"
+        And the "<ProblemType>" problem displays a "answered" answer
+        And a general notification is shown
+        """
+        # Make sure we're looking at the right problem
+        self.problem_page.wait_for(
+            lambda: self.problem_page.problem_name == self.problem_name,
+            "Make sure the problem is on the page"
+        )
+
+        # Answer the problem
+        self.answer_problem(correctness="correct")
+        self.assert_correctness_withheld()
+
+    @attr(shard=7)
+    def test_answer_incorrect_withheld(self):
+        """
+        Scenario: I can answer a problem incorrectly, and correctness is withheld.
+        Given External graders respond "incorrect"
+        and I am viewing a "<ProblemType>" problem that shows the correctness "never"
+        When I answer a "<ProblemType>" problem "incorrectly"
+        Then my "<ProblemType>" answer is marked "answered"
+        And the "<ProblemType>" problem displays a "answered" answer
+        And a general notification is shown
+        """
+        # Make sure we're looking at the right problem
+        self.problem_page.wait_for(
+            lambda: self.problem_page.problem_name == self.problem_name,
+            "Make sure the problem is on the page"
+        )
+
+        # Answer the problem
+        self.answer_problem(correctness="incorrect")
+        self.assert_correctness_withheld()
```

Comment:
@pomegranited Would it be possible to add a test for a partially correct answer? I'm not really familiar with the acceptance test code, so I'm not sure how much additional effort this would require. If it goes beyond essentially duplicating the test above and changing `self.answer_problem(correctness="incorrect")` to something like `self.answer_problem(correctness="partially-correct")`, it would be fine not to go down that particular rabbit hole I think.